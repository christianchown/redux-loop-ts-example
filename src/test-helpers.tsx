import * as React from 'react';
import * as shallowRenderer from 'react-test-renderer/shallow';
import { All, initialState } from './reducers/index';
import { Provider } from 'react-redux';
import * as deepRenderer from 'react-test-renderer';
import { createStore, compose } from 'redux';
import { install, StoreCreator } from 'redux-loop';
import { reducer } from './reducers/index';

const enhancedCreateStore = createStore as StoreCreator;

const enhancer = compose(install());
const store = enhancedCreateStore(reducer, initialState, enhancer);
// tslint:disable-next-line no-any
type TestComponent = React.ReactElement<any>;

export function renderComponentShallow(component: TestComponent) {
  return shallowRenderer.createRenderer().render(component);
}

export function renderComponent(
  component: TestComponent,
  state: All = initialState
) {
  return deepRenderer.create(<Provider store={store}>{component}</Provider>);
}
