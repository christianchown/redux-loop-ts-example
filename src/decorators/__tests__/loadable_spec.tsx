import * as React from 'react';
import { renderComponentShallow } from '../../test-helpers';
import loadable from '../loadable';

interface Props {
  isLoading: boolean;
}

const TestComponent = () => <div>OK</div>;
const LoadableTestComponent = loadable<Props>(props => props.isLoading)(
  TestComponent
);

describe('decorators/loadable', () => {
  it('sets friendly .displayName', () => {
    expect(LoadableTestComponent.displayName).toEqual(
      'Loadable(TestComponent)'
    );
  });

  it('maps props to loading', () => {
    expect(
      renderComponentShallow(<LoadableTestComponent isLoading={true} />)
    ).toMatchSnapshot();
  });
});
