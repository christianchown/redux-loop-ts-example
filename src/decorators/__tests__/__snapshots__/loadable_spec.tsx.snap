// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decorators/loadable maps props to loading 1`] = `
<div>
  Just a moment, please...
</div>
`;
